README.md
# -home-arturoaiello-mysite-


history.txt
    1  cd mysite
    2  git status
    3  git add .
    4  git commit -m "ma premiere partie est entree dans la liturgie. la seconde partie se concentre vers le quatorzieme siecle. je vous salue marie"
    5  mkvirtualenv flask-tutorial --python=python3.6
    6  pip install flask flask-login flask-sqlalchemy mysql-connector-python
    7  pip freeze > requirements.txt
    8  git status
    9  git add .
   10  git commit -m "je salue a tous mes amis seminaristes"
   11  pip install Flask-Migrate
   12  export FLASK_APP=flask_app.py
   13  flask db init
   14  git status
   15  git add .
   16  git commit -m "bon jour tres bien voila"
   17  export FLASK_APP=flask_app.py
   18  flask db init
   19  git status
   20  git add .
   21  git commit -m "qui guerit une oreille tranchee?"
   22  export FLASK_APP=flask_app.py
   23  flask db init
   24  git status
   25  mkvirtualenv flask-tutorial --python=python3.6
   26  pip install flask flask-login flask-sqlalchemy mysql-connector-python
   27  pip freeze > requirements.txt
   28  git status
   29  git add .
   30  git commit -m "sur le chemin"
   31  pip install Flask-Migrate
   32  export FLASK_APP=flask_app.py
   33  flask db init
   34  git status
   35  git add .
   36  git commit -m "nouveau roi de france"
   37  pip freeze > requirements.txt
   38  git status
   39  export FLASK_APP=flask_app.py
   40  flask db init
   41  flask db migrate
   42  flask db stamp head
   43  pip freeze > requirements.txt
   44  git status
   45  git add .
   46  git commit -m "magnifique, ca sera un beau temoignage qui fera plaisir, les papes sont elus par un conclave"
   47  flaks db migrate
   48  flask db migrate
   49  flask db upgrade
   50  git status
   51  git add .
   52  git commit -m "tete dure ou petit brebis. de l'allemagne qui signifie poils, comment se-nomme cette chemise..sept point, sans faute"
   53  flask db migrate
   54  flask db upgrade
   55  flask shell
   56  echo "# -home-arturoaiello-mysite-" >> README.md
   57  git init
   58  git add README.md
   59  git commit -m "first commit"
   60  git remote add origin https://github.com/jakegoodwell/-home-arturoaiello-mysite-.git
   61  git push -u origin master
   62  history
   63  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="arturoaiello",
    password="Exeter22",
    hostname="arturoaiello.mysql.pythonanywhere-services.com",
    databasename="arturoaiello$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "46546546546546546546546546"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/2998065d1aec_.py
"""empty message

Revision ID: 2998065d1aec
Revises: 
Create Date: 2018-08-22 23:39:11.462505

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '2998065d1aec'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('comments',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('content', sa.String(length=4096), nullable=True),
    sa.Column('posted', sa.DateTime(), nullable=True),
    sa.Column('commenter_id', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['commenter_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('comments')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/9dc6f46807f9_.py
"""empty message

Revision ID: 9dc6f46807f9
Revises: 2998065d1aec
Create Date: 2018-08-22 23:44:17.228545

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '9dc6f46807f9'
down_revision = '2998065d1aec'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###
